cmd/argocd/commands/app.go

```diff
@@ -972,15 +977,34 @@ func NewApplicationWaitCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
 				watchOperations = true
 				watchSuspended = false
 			}
+
+			var selectedResources []argoappv1.SyncWaitOperationResource
+			if resources != nil {
+				selectedResources = []argoappv1.SyncWaitOperationResource{}
```

Comment:
Can we put this into a separate helper method since we are using it in this method and the sync method?